exercises/reduce_test.rb

```diff
@@ -15,51 +15,41 @@ def test_sum_a_list_of_numbers
   def test_subtract_list_of_numbers
     numbers = [28, 12, 38, 1, 91]
     result = numbers.reduce(0) do |difference, number|
-      # Your code goes here
+      difference -= number
     end
     assert_equal -170, result
   end
 
   def test_multiply_list_of_numbers
-    skip
     numbers = [2, 3, 5, 7]
-    # initial value is 1
-    # Your code goes here
+    result = numbers.reduce(1) { |acc, num| acc *= num }
     assert_equal 210, result
   end
 
   def test_capitalize_keywords_in_phrase_one_fish_two_fish_red_fish_blue_fish
     keywords = ["fish", "blue"]
     result = keywords.reduce("one fish two fish red fish blue fish") do |init, word|
-      if init == word
-        word.upcase
+      init.gsub(word, word.upcase) 
       end
-    end
     assert_equal 'one FISH two FISH red FISH BLUE FISH', result
   end
 
   def test_divide_560_by_a_bunch_of_numbers
-    skip
     numbers = [2, 2, 2, 5, 7]
-    # initial value is 560
-    # Your code goes here
+    result = numbers.reduce(560) { |init, num| init /= num; init }
     assert_equal 2, result
   end
 
   def test_subtract_smallest_values_from_100
-    skip
     elements = [[8, 5, 3], [1, 9, 11], [4, 7, 2], [19, 34, 6]]
-    # initial value is 100
-    # Your code goes here
+    result = elements.reduce(100) { |init, elms| init -= elms.sort[0] }
     assert_equal 88, result
   end
 
   def test_add_all_the_second_values_together
-    skip
     elements = [["a", 1], ["b", 9], ["c", 21]]
-    # initial value is 0
-    # Your code goes here
+    result = elements.reduce(0) { |acc, elms| acc += elms[1] }
```

Comment:
Useless assignment to variable - `acc`. Use just operator `+`.<br>Prefer `inject` over `reduce`.